### PR TITLE
add canary example for tensorflow sample

### PR DESCRIPTION
**Pull request by animeshsingh**
**What this PR does / why we need it**:

Adds the KFService diagram to main README, as well as updates Tensorflow sample for showing canary rollout

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/123)

### CODE REVIEW

File: docs/samples/tensorflow/README.md [hunk 1]
```diff
@@ -57,4 +57,64 @@ Expected Output
...
+
+You can check for successful installation using the following command
+
+```
+knctl ingress list
+
+Ingresses
+
+Name                    Addresses  Ports         Age
+knative-ingressgateway  x.x.x.x    80,443,32400  18h
```
Comment (yuzisun) [inline docs/samples/tensorflow/README.md]: `knative-ingressgateway` is already deprecated, change to `istio-ingressgateway`?

Comment (animeshsingh) [reply docs/samples/tensorflow/README.md]: removed that check.

File: .gitignore [hunk 2]
```diff
@@ -28,3 +28,4 @@ bin
 
 # development overlay for image patch not to be checked in
 config/overlays/development/manager_image_patch.yaml
+.DS_Store
```
Comment (ellis-bigelow) [inline .gitignore]: *damn you Tim Cook*

File: docs/samples/tensorflow/README.md [hunk 3]
```diff
@@ -57,4 +57,49 @@ Expected Output
...
+## Canary Rollout
+
+To test a canary rollout, you can use the tensorflow-canary.yaml 
+
+Apply the CRD
+```
+kubectl apply -f tensorflow-canary.yaml 
+```
+
+To verify if your traffic split percenage is applied correctly, you can use the following command using the [Knative CLI](#knative-cli)
```
Comment (ellis-bigelow) [inline docs/samples/tensorflow/README.md]: Thanks to our new printing, this is now available in:
```kubectl get kfservices```

File: docs/samples/tensorflow/tensorflow-canary.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+apiVersion: "serving.kubeflow.org/v1alpha1"
```
Comment (ellis-bigelow) [inline docs/samples/tensorflow/tensorflow-canary.yaml]: Can we leave this in the canary samples?

Comment (animeshsingh) [reply docs/samples/tensorflow/tensorflow-canary.yaml]: this is showing TF e-2-e if somebody lands on TF sample. The one in canary samples is using the fake model. I plan to extend and continue making the TF example complete from all different scenarios in future

File: docs/samples/tensorflow/README.md (hunk 3 above)
Comment (animeshsingh) [reply docs/samples/tensorflow/README.md]: Ok will add it as default option to use.

Comment (animeshsingh) [reply docs/samples/tensorflow/README.md]: Done

**Comment by animeshsingh** [comment]
Seems the merge here got a bit messed up - will close this and submit as a fresh one.

**Pull request closed without merging**